### DISCUSSION TITLE
Charmstone doesn't bring all windows of app to foreground?

### DISCUSSION THREAD

**Discussion by dylanhand** [rxhanson/Charmstone-Community#43] [Q&A]
Just installed Charmstone and really like the philosophy behind it!

However I noticed that it doesn't foreground all windows of the selected app, just the one that was last used.

Is this intentional? Is there a way to override it?

I would have expected it to work like `cmd`+`tab`, which brings all windows of the selected app forward :)

**Comment by rxhanson** [comment]
Thanks for trying Charmstone!

This is sort of a weird way that macOS works, and there's oddly not a good way to do this without using the Accessibility API (which would require Charmstone to be outside of the App Store). Personally, I check the box in Charmstone for "Hide app if frontmost". Unhiding apps using Charmstone will bring all windows to the foreground, although this is a different workflow from what you might be looking for.

**Reply by dylanhand** [reply]
Thanks for the quick reply!

I understand - too bad they don't make it easier.

It sounds like the workflow requires 3 steps:
1. Select the app of interest (which brings only the last used window to foreground)
2. Hide the app of interest
3. Unhide the app of interest (which brings all windows forward)

Is that right?

I'm new enough to Charmstone that I'm not sure it's really a problem yet -- maybe I'll find it's not 😅

**Reply by rxhanson** [reply]
Yes, that is the workflow. It can take a little getting used to, but this is what I use in some scenarios. It doesn't solve the scenario where you don't want to hide the app, unfortunately.

**Reply by noahkiss** [reply]
Ryan, do you think a workaround here could be to send "hide all" and then "show" commands in succession? If that worked, three different ways to allow users to use this which I can think of:

- Checkbox which makes this functionality always run (always show all windows of an app)
- Modifier key which, when held, shows all windows
- Outer layer modifier, similar to "switch to last app", which shows all windows for the corresponding inner app (this would be the most flexible option IMO)

**Reply by rxhanson** [reply]
@noahkiss thanks for the idea! I tried this out, and it does work but it is a little bit jarring to have the app hide then show if it is visible. At some point, I am going to diverge the App Store and non-App Store versions of the app since there are a lot of features that I cannot add in the App Store version. I am thinking that I would prefer to add it in the better way in the non-App Store version. I'll keep this in mind as an option for the App Store version, although it might not make it in as something configurable in the UI.

**Reply by noahkiss** [reply]
ah, too bad that it's jarring, I worried about that if it couldn't happen fast enough. I appreciate you trying it out, at least! Yeah, it would be awesome to see it as a plist option in some future release.

**Comment by rxhanson** [comment]
I figured this out for both in and out of the App Store, and in v1.38 there's a checkbox to allow bringing all windows front. Closing this out, but let me know if there's any issues with it.
